src/main/java/stirling/software/SPDF/config/EndpointInterceptor.java

```diff
@@ -1,28 +1,225 @@
 package stirling.software.SPDF.config;
 
+import java.security.Principal;
+import java.util.Collection;
+import java.util.List;
+
 import org.springframework.stereotype.Component;
 import org.springframework.web.servlet.HandlerInterceptor;
 
 import jakarta.servlet.http.HttpServletRequest;
 import jakarta.servlet.http.HttpServletResponse;
+import jakarta.servlet.http.HttpSession;
 
 import lombok.extern.slf4j.Slf4j;
 
+import stirling.software.SPDF.config.interfaces.SessionsInterface;
+import stirling.software.SPDF.config.interfaces.SessionsModelInterface;
+
 @Component
 @Slf4j
 public class EndpointInterceptor implements HandlerInterceptor {
 
     private final EndpointConfiguration endpointConfiguration;
+    private final SessionsInterface sessionsInterface;
 
-    public EndpointInterceptor(EndpointConfiguration endpointConfiguration) {
+    public EndpointInterceptor(
+            EndpointConfiguration endpointConfiguration, SessionsInterface sessionsInterface) {
         this.endpointConfiguration = endpointConfiguration;
+        this.sessionsInterface = sessionsInterface;
     }
 
     @Override
     public boolean preHandle(
             HttpServletRequest request, HttpServletResponse response, Object handler)
             throws Exception {
+        HttpSession session = request.getSession(false);
+        if (session == null) {
```

Comment:
If you're going to check whether the session is null, you might as well just use `HttpSession session = request.getSession(true);` and skip the null check